lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -769,19 +770,19 @@ pools spl =
 listPools
     :: StakePoolLayer IO
     -> Handler [ApiStakePool]
-listPools spl = liftHandler (map mkApiStakePool <$> listStakePools spl)
+listPools spl =
+    liftHandler $ map mkApiStakePool <$> listStakePools spl
   where
     mkApiStakePool
-        :: ( W.PoolId,
-            (Quantity "lovelace" Word64, Quantity "block" Natural)
-            )
+        :: StakePool
         -> ApiStakePool
-    mkApiStakePool (pool, ((Quantity stake), blocks)) =
-        -- TODO: Make sure the stake-types match (Word64 vs Natural)
-        -- to avoid unwrapping Quantity.
+    mkApiStakePool StakePool{poolId,stake,production,apparentPerformance} =
         ApiStakePool
-            (ApiT pool)
-            (StakePoolMetrics (Quantity $ fromIntegral stake) blocks)
+            (ApiT poolId)
+            (ApiStakePoolMetrics
+                (Quantity $ fromIntegral $ getQuantity stake)
+                (Quantity $ fromIntegral $ getQuantity production))
+            (min 1 apparentPerformance)
```

Comment:
Think this `min 1` deserves an explanation / motivation

Reply:
Indeed!

Reply:
Fixed in: https://github.com/input-output-hk/cardano-wallet/pull/924/commits/0f795d435d73d437c7358b5c61bd1e961aa55d75

I've moved the clamping to be done earlier (in the `calculatePerformance` function) and also, fixed an issue with the averaging (I was counting "unknown epochs" for which we have no data, I've updated my comment above, the calculated performances look better :))

I've also extended the existing integration test to check for the stake pool performance indicator.